vllm/core/block_manager.py

```diff
@@ -435,6 +474,15 @@ def get_num_free_gpu_blocks(self) -> int:
     def get_num_free_cpu_blocks(self) -> int:
         return self.cpu_allocator.get_num_free_blocks()
 
+    def get_num_cached_blocks(self, seq: Sequence) -> int:
+        return self.gpu_allocator.get_num_cached_blocks(seq)
+
+    def get_num_computed_blocks(self, seq: Sequence) -> int:
+        return self.gpu_allocator.get_num_computed_blocks(seq)
+
+    def get_num_computed_tokens(self, seq: Sequence) -> int:
+        return self.block_size * self.get_num_computed_blocks(seq)
```

Comment:
Q: Does this overestimate the number of computed tokens? (as it is not guaranteed all slots in the block has token allocated)? Can you write a docstring that explains the behavior here? (Either it can be overestimated OR if we have a certain assumption it should be documented)